### PR TITLE
gha: Enable SEV-SNP tests on main

### PR DESCRIPTION
**Pull request by fitzthum**
Enable SNP tests. Skip the CPU constraints tests due to hotplug limitations.

### CODE REVIEW

File: tests/integration/kubernetes/k8s-cpu-ns.bats [hunk 1]
```diff
@@ -12,6 +12,7 @@ setup() {
 	[ "${KATA_HYPERVISOR}" == "firecracker" ] && skip "test not working see: ${fc_limitations}"
 	[ "${KATA_HYPERVISOR}" == "dragonball" ] && skip "test not working see: ${dragonball_limitations}"
 	[ "${KATA_HYPERVISOR}" == "qemu-tdx" ] && skip "TEEs do not support memory / CPU hotplug"
+	[ "${KATA_HYPERVISOR}" == "qemu-snp" ] && skip "Memory / CPU hotplug support not implemented for SEV-SNP"
```
Comment (fidencio) [inline tests/integration/kubernetes/k8s-cpu-ns.bats:15]: Could you extend the conditional and / or use the same message for both TDX / SNP / SEV?

File: tests/integration/kubernetes/k8s-cpu-ns.bats [hunk 2]
```diff
@@ -29,6 +30,7 @@ setup() {
...
```
Comment (fidencio) [inline tests/integration/kubernetes/k8s-cpu-ns.bats:33]: Same here.

File: tests/integration/kubernetes/k8s-cpu-ns.bats [hunk 3]
```diff
@@ -74,6 +76,7 @@ teardown() {
...
```
Comment (fidencio) [inline tests/integration/kubernetes/k8s-cpu-ns.bats:79]: Same here.

**Review by fidencio — CHANGES_REQUESTED** [review]
Tests passed, yay!

I've left one comment, and I think you could, as a different commit but still part of the same PR, enable the SEV tests as well.

WDYT?

**Review by fidencio — CHANGES_REQUESTED** [review]
Tests passed, yay!

I've left one comment, and I think you could, as a different commit but still part of the same PR, enable the SEV tests as well.

WDYT?

**Review by fidencio — APPROVED** [review]
lgtm, thanks @fitzthum!

**Comment by fitzthum** [comment]
Passing with SEV and SNP. Not sure about issue with `run-k8s-tests-on-aks / run-k8s-tests (qemu)`. Is that common? 

Looks like a lot of the Jenkins jobs are getting stuck looking for apt packages.

**Comment by fidencio** [comment]
> Passing with SEV and SNP. Not sure about issue with `run-k8s-tests-on-aks / run-k8s-tests (qemu)`. Is that common?

Common is a hard word. :-)
It's been flaky and we need to debug that.  Whenever you see those errors, please, open an issue pointing to which test is failing and that will help me to debug it further.

> Looks like a lot of the Jenkins jobs are getting stuck looking for apt packages.

Yeah, Yesterday was not a good day for `apt`.  Let's hope Today it does better.

**Comment by fidencio** [comment]
@fitzthum, I've solved the conflicts, as those were originated by one of my PRs that got merged first, and force-pushed to your branch.

**Review by stevenhorsman — APPROVED** [review]
LGTM assuming that the tests currently running pass. Thanks @fitzthum

**Comment by fidencio** [comment]
For the first time, all the tests were executed in the TEEs we support on x86_64. \o/

**Pull request merged**